addons/crm_survey/models/survey_user_input.py

```diff
@@ -0,0 +1,162 @@
+            lead_name_prefix = _('New')
+            user_name_for_lead_name = user_input_lines_lead_data["user_nickname"] or user_input_lines_lead_data["public_user_mail"]
+            if user_name_for_lead_name:
+                lead_name_prefix = _("%s's", user_name_for_lead_name)
+        else:
+            lead_name_prefix = _("%s's", username)
+
+        lead_values = {
+            'contact_name': username or user_input_lines_lead_data["user_nickname"],
+            'description': user_input_lines_lead_data["description"],
+            'name': f"{lead_name_prefix} {_('live session results')}" if self.is_session_answer else f"{lead_name_prefix} {_('survey results')}",
+            'user_id': survey_responsible.id,
+        }
+
+        # Associate lead to the existing partner when known. Either because the person is connected
+        # or because they received the survey with a unique token (by email for example).
+        if self.partner_id:
+            lead_values['partner_id'] = self.partner_id.id
+        elif user_input_lines_lead_data["public_user_mail"]:  # Save email field answer otherwise
+            lead_values['email_from'] = user_input_lines_lead_data["public_user_mail"]
+
+        return lead_values
+
+    def _prepare_lead_values_from_user_input_lines(self):
+        """ This method will prepare necessary lead values (in a dictionary), from the
+            content of user input lines, to create a lead: it will format the lead
+            description and get user's nickname and his email, if they're provided.
+            To write the description (notes in CRM) in HTML format, there are 5 cases:
+                - Suggested answers question: <li>Question — Answer 1, Answer 3</li>
+                - Matrix question:  <li>
+                                        Question
+                                        <br>&esmpLine label 1 — Answer 1
+                                        <br>&esmpLine label 1 — Answer 3
+                                        <br>&esmpLine label 4 — Answer 2
+                                    </li>
+                - Long text question:   <li>
+                                            Question
+                                            <br>&esmpText line 1
+                                            <br>&esmpText line 2
+                                            <br>&esmpText line 3
+                                        </li>
+                - Other types: <li>Question — Answer</li>
+                - Skipped question: <li>Question — <i>Skipped</i></li>
+
+            Note: "<br/>&emsp;" is used to get a line break with indentation.
+        """
+        self.ensure_one()
+
+        answers_grouped_by_question = self.user_input_line_ids.grouped('question_id')
+        html_input_lines = []
+        user_nickname, public_user_mail = "", ""
+        for question_id, input_lines in answers_grouped_by_question.items():
+            answer_description, last_row_id = "", ""
```

Comment:
unlikely, but if at review you are asked to be certain that we group matrix inputs by row, this should be done here (input_lines = input_lines.sorted...)